### PR TITLE
Fixed #24127 -- Changed the default current_app to the current namespace.

### PR DESCRIPTION
**Pull request by knbk**
Changed the `url` template tag to use `request.resolver_match.namespace` as a
default for the `current_app` argument if `request.current_app` is not set.

https://code.djangoproject.com/ticket/24127

### CODE REVIEW

**Comment by knbk** [comment]
It doesn't hurt to properly document `request.current_app`, so I made #4920. That should fix the doc error here.

File: docs/releases/1.9.txt [hunk 1]
```diff
@@ -742,6 +742,12 @@ Miscellaneous
   be serialized contains any control characters not allowed in the XML 1.0
   standard, the serialization will fail with a :exc:`ValueError`.
 
+* If both :attr:`request.current_app <django.http.HttpRequest.current_app>`
```
Comment (timgraham) [inline docs/releases/1.9.txt]: If neither a nor b are set, ...

File: docs/topics/http/urls.txt [hunk 2]
```diff
@@ -682,8 +687,11 @@ the fully qualified name into parts and then tries the following lookup:
       :class:`~django.template.RequestContext` that is used to render a
       template.
 
-   The current application can also be specified manually as an argument
-   to the :func:`~django.core.urlresolvers.reverse` function.
+   .. versionchanged:: 1.9
+
+      In previous versions of Django, the :ttag:`url` template tag did not use
+      the namespace of the currently resolved view, but you had to explicitly
```
Comment (timgraham) [inline docs/topics/http/urls.txt]: but -> and

**Comment by timgraham** [comment]
tests to be added?

File: docs/releases/1.9.txt [hunk 3]
```diff
@@ -892,6 +892,11 @@ Miscellaneous
 * :class:`~django.forms.CharField` now strips input of leading and trailing
   whitespace by default. This can be disabled by setting the new
   :attr:`~django.forms.CharField.strip` argument to ``False``.
+* If neither :attr:`request.current_app <django.http.HttpRequest.current_app>`
```
Comment (timgraham) [inline docs/releases/1.9.txt]: Add blank line above

File: docs/topics/http/urls.txt [hunk 4]
```diff
@@ -682,8 +687,11 @@ the fully qualified name into parts and then tries the following lookup:
       :class:`~django.template.RequestContext` that is used to render a
       template.
 
-   The current application can also be specified manually as an argument
-   to the :func:`~django.core.urlresolvers.reverse` function.
+   .. versionchanged:: 1.9
+
+      In previous versions of Django, the :ttag:`url` template tag did not use
+      the namespace of the currently resolved view, and you had to explicitly
```
Comment (timgraham) [inline docs/topics/http/urls.txt]: chop comma

Comment (timgraham) [inline docs/topics/http/urls.txt]: Also, farther down on this page there's a mention about setting `self.request.current_app` in class based views which can probably be removed.

**Comment by timgraham** [comment]
Don't forget to tick the appropriate flags on the ticket when updating so the ticket appears in the review queue. Thanks!

**Comment by timgraham** [comment]
merged in bc7923beffea3c51271935cad1371a67c5fcc71b, thanks!

**Pull request closed without merging**